### ISSUE TITLE
Continuation of #696 Disconnecting relays

### ISSUE THREAD

**Issue by darkshadow543** [Ellpeck/ActuallyAdditions#719]
I am using the current version of Actually Additions, and whenever I reenter my world, not from the nether, but from the menu, there are always a few random disconnections.  The last time this happened, half of them disconnected.

Edit: its appears to occur after the I use Bed Bugs mod to force me back to the survival menu

**Comment by darkshadow543** [comment]
PS. It is getting on my nerves.  #696

**Comment by Ellpeck** [comment]
Please record a video of the issue happening and also send the mod list and the world you are using.
Also, please try to be a little less rude. We're both trying to sort out this issue and it's not helping anything.

**Comment by darkshadow543** [comment]
I reclogged to see what would happen, and sure enough, some broke.  No the bed bug did not happen.
Before Reloging
![2017-03-29_12 00 38](https://cloud.githubusercontent.com/assets/14943673/24464932/74a41ac4-1479-11e7-96dc-7e532aa4fb53.png)
![2017-03-29_12 00 43](https://cloud.githubusercontent.com/assets/14943673/24465007/a690a0a2-1479-11e7-8c26-d633349724d0.png)
![2017-03-29_12 00 48](https://cloud.githubusercontent.com/assets/14943673/24465014/abff3b52-1479-11e7-85c1-3a81255c98fb.png)
After Reloging
![2017-03-29_12 14 29](https://cloud.githubusercontent.com/assets/14943673/24465035/bc7f956c-1479-11e7-9599-b808f526c190.png)
![2017-03-29_12 14 38](https://cloud.githubusercontent.com/assets/14943673/24465039/bec593f8-1479-11e7-8125-4e1179aee519.png)
![2017-03-29_12 15 01](https://cloud.githubusercontent.com/assets/14943673/24465041/c4d0bdcc-1479-11e7-93e9-c3e062c22cad.png)
And after Reloging again, more were disconnected:
![2017-03-29_12 19 41](https://cloud.githubusercontent.com/assets/14943673/24465140/0f904828-147a-11e7-8762-95b3fb58de06.png)
![2017-03-29_12 19 44](https://cloud.githubusercontent.com/assets/14943673/24465148/1319aa98-147a-11e7-80f4-c001eac71e2d.png)
![2017-03-29_12 19 49](https://cloud.githubusercontent.com/assets/14943673/24465152/1571b3d0-147a-11e7-85f3-6573f06ae4a9.png)

**Comment by Ellpeck** [comment]
Please record a video of the entire thing happening.

**Comment by darkshadow543** [comment]
I tried a video, it's to big, so up to youtube it goes.  Even one of my farms lasers broke.

**Comment by darkshadow543** [comment]
https://youtu.be/snQsoD9mh_o 
Link to video, sorry about not using power lasers in the base:{(

**Comment by Ellpeck** [comment]
Can you figure out what exactly might be causing the issue?
Do you know of a certain event that might trigger it? Or does it happen all the time?

**Comment by darkshadow543** [comment]
Happens all the time, if it crashes from a bug with refined storage, it does not disconnect.  If you relog, then it disconnects, and it is only upon reclogging that they disconnect.  Tried unloading the base chunks and generating new chunks, and nothing happened.  Some always disconnect, I think it may have to do with the way things are stored in the .dat file, and may not be your fault, but the game codes fault.

**Comment by Ellpeck** [comment]
_This is a note to my future self if I don't manage to get this done today:_
I'll try moving the disconnection code away from the TileEntity and to the Block's onRemove method, and then I'll send you a special jar to try out if the bug still happens with that.

**Comment by Ellpeck** [comment]
Could you try reproducing it with [this version](http://puu.sh/v3tyO/67a4dc526a.jar)? It's called r106, but I actually made some changes. Back up your world tough, this might break stuff.

**Comment by darkshadow543** [comment]
I happened to a few energy relays on my farm as well, but will do.

**Comment by Ellpeck** [comment]
Can you reproduce it in a different world?

**Comment by Ellpeck** [comment]
Could you try again with [this version](http://puu.sh/v4vmt/524d3a5954.jar) and see if it still happens?
Please send the full log of a game instance that this happened with. No video required this time, though.

**Comment by darkshadow543** [comment]
I doubt its your mod, I just realized that the laser relay that connects my quarry to the crusher never broke, not once,  however, if any of the relays broke, it was the ones that were connected to the refined storage system.  All of the ones connected to the Refined Storage system are the ones that broke.

**Comment by darkshadow543** [comment]
Or it could be location, that causes it.

**Comment by darkshadow543** [comment]
Went into another world, set up everything except location, did not happen in that world not using the new version yet.  Maybe farmers are interacting with it in an odd way.

**Comment by darkshadow543** [comment]
[2017-03-31-7.log.gz](https://github.com/Ellpeck/ActuallyAdditions/files/887318/2017-03-31-7.log.gz)

**Comment by darkshadow543** [comment]
There is a log of me connecting the relay and logging off using the new version.

**Comment by darkshadow543** [comment]
I program as well and a CIT student at the current moment, perhaps it has to do with location? -70 - -110 on both x and z axis is about the range of coordanites, some are on the outside of this.

**Comment by darkshadow543** [comment]
Perhaps used an = in an if statement somewhere?

**Comment by darkshadow543** [comment]
Two Locations for bug, maybe
TileEntityLaserRelay.java 
Line 52: if(type == NBTType.SYNC){

            ActuallyAdditionsAPI.connectionHandler.removeRelayFromNetwork(this.pos, this.world);
or any of the following code
or in the method saveDataOnChangeOrWorldStart() in the relay classes.

Or perhaps it is a conflict of storage between the information of two relays.

**Comment by Ellpeck** [comment]
Thanks, but you don't need to assess my code.

**Comment by Ellpeck** [comment]
I know this is quite a huge task, but could you try setting the Laser Connections up in a new world in roughly the same coordinates, or at least the same +/- quadrant, and see if they act weirdly with the newest version I sent you?

**Comment by darkshadow543** [comment]
One more thing I'd like to report, Refined Storage would crash on login sometimes.  Throwing an array out of bounds exception.  Maybe this is related to the problem.

**Comment by darkshadow543** [comment]
It happens upon loading, with advanced item relays, yes in the position.  I was extremely lucky and saw one of the relays load in connected before becoming, this may help.

**Comment by darkshadow543** [comment]
It saves correctly, but does not load properly, it loads connected and for some reason, it dissconects

**Comment by Ellpeck** [comment]
So what now? Did you do what I told you to do?
What was the result? Is the issue fixed?

**Comment by darkshadow543** [comment]
One more Bug, connecting lasers which are part of the same network does not give the error message, it give the laser connected message

**Comment by darkshadow543** [comment]
Traveling to the next quadrant now

**Comment by Ellpeck** [comment]
Connecting lasers which are part of the same network disconnects them, that's not a bug.

**Comment by Ellpeck** [comment]
Could you please collect the information I ask you to connect and then post all of it in one comment? This is not a chat room.

**Comment by darkshadow543** [comment]
It was in the same +/- quadrant (-x, -z) not all the same positions, but they did disconnect using the latest version, in the same area.

**Comment by Ellpeck** [comment]
_sigh_
Which world is this in? It would be nice if you would give full information about what you're doing and how it turns out instead of being all over the place. It's really hard to follow.

**Comment by darkshadow543** [comment]
Sorry, I have ADHD, its how I think, I can not help it.

**Comment by Ellpeck** [comment]
Try removing mods to see if the issue is fixed eventually. If so, try finding out which mod causes it.

**Comment by darkshadow543** [comment]
After deleting Armor Plus, it has not happened again.  I used the mod list the other guy posted to narrow it down.

**Comment by darkshadow543** [comment]
Cant pinpoint the mod that is causing the issue, Sorry, I tried.
